crates/bevy_app/src/app.rs

```diff
@@ -30,21 +32,36 @@ use bevy_utils::tracing::info_span;
 /// }
 /// ```
 pub struct App {
-    pub world: World,
-    pub runner: Box<dyn Fn(App)>,
-    pub schedule: Schedule,
+    pub world: ManuallyDrop<World>,
+    pub runner: ManuallyDrop<Box<dyn Fn(App)>>,
+    pub schedule: ManuallyDrop<Schedule>,
 }
 
 impl Default for App {
     fn default() -> Self {
         Self {
             world: Default::default(),
             schedule: Default::default(),
-            runner: Box::new(run_once),
+            runner: ManuallyDrop::new(Box::new(run_once)),
         }
     }
 }
 
+impl Drop for App {
+    fn drop(&mut self) {
+        unsafe {
+            drop_app(self);
+        }
+    }
+}
+
+// Outlined to ensure that a lot of drop glue is codegened inside bevy_app instead of the game.
+unsafe fn drop_app(app: &mut App) {
```

Comment:
Add a `// SAFE` comment explaining why this is safe to use.
It's pretty obvious why it is safe, but I'd argue that every use of `unsafe` needs to have a `// SAFETY` comment to go along with it.

Reply:
More specifically the format is:

``` rust
/// # Safety
/// <explination>
unsafe fn drop_ass(app: &mut App) {}
```